cpp/src/arrow/buffer.h

```diff
@@ -54,7 +54,11 @@ class ARROW_EXPORT Buffer {
   ///
   /// \note The passed memory must be kept alive through some other means
   Buffer(const uint8_t* data, int64_t size)
-      : is_mutable_(false), data_(data), size_(size), capacity_(size) {}
+      : is_mutable_(false),
+        data_(data),
+        mutable_data_(nullptr),
```

Comment:
We need to use the NULLPTR macro here

Reply:
Can you expand on the NULLPTR macro? Does it do something more then `nullptr`?

Reply:
`nullptr` is incompatible C++/CLI, which I believe is a way for C# code to link to C++ libraries. see https://msdn.microsoft.com/en-us/library/4ex65770.aspx